include/seqan3/range/container/bitcompressed_vector.hpp

```diff
@@ -0,0 +1,1114 @@
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Equals the alphabet_type.
+    using value_type        = alphabet_type;
+    //!\brief A proxy type that enables assignment, if the underlying data structure also provides a proxy.
+    using reference         = std::conditional_t<std::is_lvalue_reference_v<reference_t<data_type>>,
+                                                 reference_t<data_type>,
+                                                 reference_proxy_type>;
+    //!\brief Equals the alphabet_type / value_type.
+    using const_reference   = alphabet_type;
+    //!\brief The iterator type of this container (a random access iterator).
+    using iterator          = detail::random_access_iterator<bitcompressed_vector>;
+    //!\brief The const_iterator type of this container (a random access iterator).
+    using const_iterator    = detail::random_access_iterator<bitcompressed_vector const>;
+    //!\brief A signed integer type (usually std::ptrdiff_t)
+    using difference_type   = ranges::difference_type_t<data_type>;
+    //!\brief An unsigned integer type (usually std::size_t)
+    using size_type         = ranges::size_type_t<data_type>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    bitcompressed_vector() = default;
+    constexpr bitcompressed_vector(bitcompressed_vector const &) = default;
+    constexpr bitcompressed_vector(bitcompressed_vector &&) = default;
+    constexpr bitcompressed_vector & operator=(bitcompressed_vector const &) = default;
+    constexpr bitcompressed_vector & operator=(bitcompressed_vector &&) = default;
+    ~bitcompressed_vector() = default;
+
+    /*!\brief Construct/assign from a different range.
+     * \tparam other_range_t The type of range to be inserted; must satisfy seqan3::input_range_concept and
+     *                       seqan3::common_reference_concept<value_type_t<other_range_t>, value_type>.
+     * \param[in]      range The sequences to construct/assign from.
+     *
+     * ### Complexity
+     *
+     * Linear in the size of `range`.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    template <typename other_range_t>
+    explicit bitcompressed_vector(other_range_t && range)
+    //!\cond
+        requires input_range_concept<other_range_t> && has_same_value_type_v<other_range_t>
+    //!\endcond
+    {
+        insert(cend(), ranges::begin(range), ranges::end(range));
+    }
+
+    /*!\brief Construct/assign with `count` times `value`.
+     * \param[in] count Number of elements.
+     * \param[in] value The initial value to be assigned.
+     *
+     * ### Complexity
+     *
+     * In \f$O(count*value)\f$.
+     *
+     * ### Exceptions
+     *
+     * Strong exception guarantee (no data is modified in case an exception is thrown).
+     */
+    bitcompressed_vector(size_type const count, value_type const value)
```

Comment:
you could also  use constructor delegation:
```cpp
bitcompressed_vector(size_type const count, value_type const value) : bitcompressed_vector(view::generate_n(value, count))
{}
```
Thus, checks like an empty range and so on need to be done only once.